File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/medlemskapsperiode/dto/MedlemskapsperiodeDto.java
package no.nav.melosys.tjenester.gui.medlemskapsperiode.dto;

import java.time.LocalDate;

import no.nav.melosys.domain.Medlemskapsperiode;
import no.nav.melosys.domain.kodeverk.InnvilgelsesResultat;
import no.nav.melosys.domain.kodeverk.Medlemskapstyper;
import no.nav.melosys.domain.kodeverk.Trygdedekninger;

public record MedlemskapsperiodeDto(long id,
                                    String arbeidsland,
                                    LocalDate fomDato,
                                    LocalDate tomDato,
                                    InnvilgelsesResultat innvilgelsesResultat,
                                    Trygdedekninger trygdedekning,
                                    Medlemskapstyper medlemskapstype) {

    public static MedlemskapsperiodeDto av(Medlemskapsperiode medlemskapsperiode) {
        return new MedlemskapsperiodeDto(
            medlemskapsperiode.getId(),
            medlemskapsperiode.getArbeidsland(),
            medlemskapsperiode.getFom(),
            medlemskapsperiode.getTom(),
            medlemskapsperiode.getInnvilgelsesresultat(),
            medlemskapsperiode.getTrygdedekning(),
            medlemskapsperiode.getMedlemskapstype()
        );
    }
}

File: integrasjon/src/main/kotlin/no/nav/melosys/integrasjon/dokgen/dto/innvilgelseftrl/MedlemskapsperiodeDto.kt
package no.nav.melosys.integrasjon.dokgen.dto.innvilgelseftrl

import tools.jackson.databind.annotation.JsonSerialize
import tools.jackson.databind.ext.javatime.ser.LocalDateSerializer
import no.nav.melosys.domain.kodeverk.InnvilgelsesResultat
import no.nav.melosys.domain.kodeverk.Trygdedekninger
import java.time.LocalDate

data class MedlemskapsperiodeDto(
    @JsonSerialize(using = LocalDateSerializer::class) val fom: LocalDate,
    @JsonSerialize(using = LocalDateSerializer::class) val tom: LocalDate?,
    val trygdedekning: Trygdedekninger,
    val innvilgelsesResultat: InnvilgelsesResultat
)
